game/plugin/skills/runecrafting/src/data.kt

```diff
@@ -0,0 +1,115 @@
+
+    companion object {
+        fun findByEntranceId(id: Int): Alter? = Alter.values().find { alter ->  alter.entranceId == id }
+        fun findByPortalId(id: Int): Alter? = Alter.values().find { alter ->  alter.portalId == id }
+        fun findByCraftingId(id: Int): Alter? = Alter.values().find { alter ->  alter.craftingId == id }
+    }
+}
+
+enum class Rune(val id: Int, val alter: Alter, val level: Int, val xp: Double) {
+    AIR_RUNE(556, Alter.AIR_ALTER, 1, 5.0),
+    MIND_RUNE(558, Alter.MIND_ALTER, 1, 5.5),
+    WATER_RUNE(555, Alter.WATER_ALTER, 5, 6.0),
+    EARTH_RUNE(557, Alter.EARTH_ALTER, 9, 6.5),
+    FIRE_RUNE(554, Alter.FIRE_ALTER, 14, 7.0),
+    BODY_RUNE(559, Alter.BODY_ALTER, 20, 7.5),
+    COSMIC_RUNE(564, Alter.COSMIC_ALTER, 27, 8.0),
+    CHAOS_RUNE(562, Alter.CHAOS_ALTER, 35, 8.5),
+    NATURE_RUNE(561, Alter.NATURE_ALTER, 44, 9.0),
+    LAW_RUNE(563, Alter.LAW_ALTER, 54, 9.5),
+    DEATH_RUNE(560, Alter.DEATH_ALTER, 65, 10.0);
+
+    companion object {
+        fun findById(id: Int): Rune? = Rune.values().find { rune ->  rune.id == id }
+        fun findByAlterId(id: Int): Rune? = Rune.values().find { rune ->  rune.alter.craftingId == id }
+    }
+
+    fun getBonus(): Double =
+            when(this) {
+                Rune.AIR_RUNE -> (floor((level / 11.0)) + 1)
+                Rune.MIND_RUNE -> (floor((level / 14.0)) + 1)
+                Rune.WATER_RUNE -> (floor((level / 19.0)) + 1)
+                Rune.EARTH_RUNE -> (floor((level / 26.0)) + 1)
+                Rune.FIRE_RUNE -> (floor((level / 35.0)) + 1)
+                Rune.BODY_RUNE -> (floor((level / 46.0)) + 1)
+                Rune.COSMIC_RUNE -> (floor((level / 59.0)) + 1)
+                Rune.CHAOS_RUNE -> (floor((level / 74.0)) + 1)
+                Rune.NATURE_RUNE -> (floor((level / 91.0)) + 1)
+                Rune.LAW_RUNE -> 1.0
+                Rune.DEATH_RUNE -> 1.0
+            }
+}
+
+enum class Talisman(val id: Int, val alter: Position) {
+    AIR_TALISMAN(1438, Position(2985, 3292)),
+    EARTH_TALISMAN(1440, Position(3306, 3474)),
+    FIRE_TALISMAN(1442, Position(3313, 3255)),
+    WATER_TALISMAN(1444, Position(3185, 3165)),
+    BODY_TALISMAN(1446, Position(3053, 3445)),
+    MIND_TALISMAN(1448, Position(2982, 3514)),
+    CHAOS_TALISMAN(1452, Position(3059, 3590)),
+    COSMIC_TALISMAN(1454, Position(2408, 4377)),
+    DEATH_TALISMAN(1456, Position(0, 0)),
+    LAW_TALISMAN(1458, Position(2858, 3381)),
+    NATURE_TALISMAN(1462, Position(2869, 3019));
+
+    companion object {
+        fun findById(id: Int): Talisman? = Talisman.values().find { talisman ->  talisman.id == id }
+    }
+
+    fun sendMessage(player: Player) {
+        if (alter.isWithinDistance(player.position, 10)) {
+            player.sendMessage("Your talisman glows brightly.");
+        } else {
+            var msg = if (player.position.y > alter.y) "North" else "South";
```

Comment:
Could this logic be moved into the `Direction` class?  It's possible to get a `Direction` from two positions, maybe those can have readable names.